gen/scim_atlassian_mu

```diff
@@ -0,0 +1,134 @@
+our $A_USER_FAMILY_NAME;          *A_USER_FAMILY_NAME =      \'urn:perun:user:attribute-def:core:lastName';
+our $A_GROUP_NAME;                *A_GROUP_NAME =            \'urn:perun:group:attribute-def:core:name';
+
+#my $domain = "\@muni.cz";
+my $domain = "\@test4mu.servicedesk.muni.cz";
+
+my $userStruc = {};
+my $groupStruc = {};
+my $membershipStruc = {};
+
+my $email = {};
+my $givenName = {};
+my $familyName = {};
+
+my $fileUsers = $DIRECTORY . "/users.scim";
+my $fileGroups = $DIRECTORY . "/groups.scim";
+
+foreach my $resourceId ($data->getResourceIds()) {
+	foreach my $groupId ($data->getGroupIdsForResource( resource => $resourceId )) {
+		processGroups $groupId, $resourceId;
+	}
+}
+
+# PREPARE USERSDATA TO JSON
+my %users = ();
+foreach my $uid (sort keys %$userStruc) {
+	my $userInfo = {};
+	$userInfo->{"email"} = $userStruc->{$uid}->{$email};
+	$userInfo->{"givenName"} = $userStruc->{$uid}->{$givenName};
+	$userInfo->{"familyName"} = $userStruc->{$uid}->{$familyName};
+	$users{$userStruc->{$uid}->{$email}} = $userInfo;
+}
+
+# PRINT USERS TO JSON
+open FILE_USERS,">$fileUsers" or die "Cannot open $fileUsers: $! \n";
+binmode(FILE_USERS);
+print FILE_USERS JSON::XS->new->utf8->pretty->canonical->encode(\%users);
+close (FILE_USERS) or die "Cannot close $fileUsers: $! \n";
+
+# PREPARE GROUPSDATA TO JSON
+my %groups;
+foreach my $gid (sort keys %$groupStruc) {
+	my $group = {};
+
+	my @members;
+	foreach my $uid (sort keys %{$membershipStruc->{$gid}}){
+		push @members, $userStruc->{$uid}->{$email};
+	}
+
+	$groups{$groupStruc->{$gid}->{$A_GROUP_NAME}} = \@members;
+}
+
+# PRINT GROUPS TO JSON
+open FILE_GROUPS,">$fileGroups" or die "Cannot open $fileGroups: $! \n";
+binmode(FILE_GROUPS);
+print FILE_GROUPS JSON::XS->new->utf8->pretty->canonical->encode(\%groups);
+close (FILE_GROUPS) or die "Cannot close $fileGroups: $! \n";
+
+perunServicesInit::finalize;
+
+##############################################################################
+#   Only subs definitions down there
+##############################################################################
+## creates structure for users.scim file
+sub processUsers {
+	my ($gid, $memberId) = @_;
+
+	my $uid = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_ID );
+	my $userFamilyName = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_FAMILY_NAME );
+	my $userGivenName = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_GIVEN_NAME );
+	my $login = $data->getUserFacilityAttributeValue( member => $memberId, facility => $data->getFacilityId, attrName => $A_USER_LOGIN );
+
+	unless (exists $userStruc->{$uid}) {
+		$userStruc->{$uid}->{$email} = $login . $domain;
+		$userStruc->{$uid}->{$familyName} = $userFamilyName;
+		$userStruc->{$uid}->{$givenName} = $userGivenName;
+	}
+
+	processMemberships $gid, $uid;
+}
+
+## creates structure for groups.scim file
+sub processGroups {
+	my $groupId = shift;
+	my $resourceId = shift;
```

Comment:
In these three functions, you are using 2 different ways how to get references on variables. I prefer to choose only one of them and be consistent in the script.